npm/design-system/src/components/FileExplorer/FileExplorer.spec.tsx

```diff
@@ -0,0 +1,130 @@
+import { mount } from '@cypress/react'
+import React from 'react'
+import { FileExplorer, FileComponentProps, FolderComponentProps } from './FileExplorer'
+import { FileNode, makeFileHierarchy, TreeNode } from './helpers/makeFileHierarchy'
+
+import styles from './FileExplorer.module.scss'
+
+const specs: Cypress.Cypress['spec'][] = [
+  {
+    relative: 'foo/bar/foo.spec.js',
+    absolute: 'Users/code/foo/bar/foo.spec.js',
+    name: 'foo/bar/foo.spec.js',
+  },
+  {
+    relative: 'bar/foo.spec.tsx',
+    absolute: 'bar/foo.spec.tsx',
+    name: 'bar/foo.spec.tsx',
+  },
+  {
+    relative: 'merp/map.spec.ts',
+    absolute: 'merp/map.spec.ts',
+    name: 'merp/map.spec.ts',
+  },
+]
+
+interface FileExplorerTestProps {
+  clickFileStub: typeof cy.stub
+  clickFolderStub: typeof cy.stub
+}
+
+function createFileExplorer (testProps: FileExplorerTestProps): React.FC {
+  return () => {
+    const [selectedFile, setSelectedFile] = React.useState<string>()
+
+    const onFileClick = (file: FileNode) => {
+      setSelectedFile(file.absolute)
+    }
+
+    const files = makeFileHierarchy(specs.map((spec) => spec.relative))
+
+    const FileComponent: React.FC<FileComponentProps> = (props) => {
+      return (
+        <div onClick={() => {
+          testProps.clickFileStub(props.item)
+          props.onClick(props.item)
+        }}>
+          {props.item.name}
+        </div>
+      )
+    }
+
+    const FolderComponent: React.FC<FolderComponentProps> = (props) => {
+      return (
+        <div onClick={() => {
+          testProps.clickFolderStub()
+          props.onClick()
+        }}>
+          {props.item.name}
+        </div>
+      )
+    }
```

Comment:
In ideal (non-test) usage, these would be statically defined at the module level, rather than inline. Alternatively, they could be memoed.

Reply:
Yep, this is just for testing purposes. We should make a new component in practice. I'll make this change in the other file where they are declared inline.